html2asketch/nodeTreeToSketchGroup.js

```diff
@@ -0,0 +1,58 @@
+import nodeToSketchLayers from './nodeToSketchLayers';
+import Group from './model/group';
+import Style from './model/style';
+
+function isNotEmpty(value) {
+  return value !== undefined && value !== null && value !== '';
+}
+
+export default function nodeTreeToSketchGroup(node, options) {
+  const bcr = node.getBoundingClientRect();
```

Comment:
If an element is `display: contents` this will always return with all `0`s. For this particular feature, you might consider a check to see if that's the case then check the contents or the `HTMLElement.shadowRoot` to see if you can't ensure there's an element.

Reply:
Oh, nice, I didn't know that! Fantastic that you figured it all out 👏 This PR is already massive though, so maybe lets add shadow DOM support in a separate one. WDYT?

Reply:
@calebdwilliams Thanks for the feedback, I am quite new to frontend technologies, at least the modern ones, so I don't know what the contents display type is, and I only heard of the shadow DOM so far, never used it. Can you please provide an example snippet of the function you have in mind?